### DISCUSSION TITLE
tasklite shell

### DISCUSSION THREAD

**Discussion by gour** [ad-si/TaskLite#84] [Ideas]
Hello,

whenever I use TW, I use it within `tasksh` which is always running in one of my terminal's tabs, so I wonder what do you think about something like `tasklitesh` ?

**Comment by ad-si** [comment]
Sounds like a good idea! However, my focus right now is on the 1.0 release, which will contain several breaking changes for the database schema and how sub-commands work (so quite some work necessary).
If you (or anyone else) opens a PR for this feature, I'd be happy to merge it!

For reference here the docs for `tasksh`: https://taskwarrior.org/docs/review/.

**Reply by gour** [reply]
> Sounds like a good idea! However, my focus right now is on the 1.0 release, which will contain several breaking changes for the database schema and how sub-commands work (so quite some work necessary). If you (or anyone else) opens a PR for this feature, I'd be happy to merge it!

OK, there is no rush for it.

**Reply by gour** [reply]
> Sounds like a good idea! However, my focus right now is on the 1.0 release, which will contain several breaking changes for the database schema and how sub-commands work (so quite some work necessary). 

How is your progress towards 1.0?

Currently I use TW and play(ed) with [Etm](https://github.com/dagraham/etm-dgraham) which is done in Python. It used `dateutil` and has no issues with recurring/repeating, but it's not as productive as TW.

Still have clunky org-mode setup and evaluating [org-gtd](https://github.com/Trevoke/org-gtd.el), but it is probably more complex (aka convoluted) in comparison with TW.

Tried briefly `smos` - nice, done in Haskell, but a bit opinionated.

I did expect to see "recurrence overhaul" in tw-3.x, but it seems there is no incentive by the main developer to work on  it. Personally, I do have neither time nor the interest to dive return to C++ or jump into Rust...

Tasklite is, otoh, a great foundation to add some possibly missing parts, written in Haskell which I could find motive to learn, it (hopefully) has proper support for repeating/recurring tasks and flexible enough to combine it with other/external tools. :sunglasses:
